pat-static/src/main/kotlin/com/rectanglel/patstatic/patterns/PatternDataManager.kt

```diff
@@ -84,7 +86,7 @@ class PatternDataManager(dataDirectory : File,
                         throw Exceptions.propagate(e)
                     }
                 }
-                .onErrorResumeNext { getPatternsFromDisk(rt) }
+                .onErrorResumeNext(getPatternsFromDisk(rt))
```

Comment:
This is causing the PatternDataManagerTest unit tests to fail.

For some reason, using the method and passing in the Flowable to return patterns from disk instead of using a lambda that returns the flowable causes issues.

I'm trying to figure out the correct syntax, but it seems rxjava2 doesn't have any documentation on onErrorResumeNext (so maybe there is a new method we should be using instead): http://reactivex.io/documentation/operators/catch.html

Reply:
Ok so I found the solution. It's because you need to pass in this lambda instead:

```
.onErrorResumeNext { _: Throwable ->  getPatternsFromDisk(rt) }
```
RxJava 2 team added a new overload that passes another lambda (don't know why) so this code above is the correct way to tell the kotlin compiler which overload to use. I don't know what using parentheses does still...